pkg/api/v1/conversion.go

```diff
@@ -223,3 +225,136 @@ func convert_v1_ReplicationControllerSpec_To_api_ReplicationControllerSpec(in *R
 	}
 	return nil
 }
+
+// The following two PodSpec conversions are done here to support ServiceAccount
+// as an alias for ServiceAccountName.
+func convert_api_PodSpec_To_v1_PodSpec(in *api.PodSpec, out *PodSpec, s conversion.Scope) error {
```

Comment:
I recommend using https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/conversion/converter.go#L111 and writing these as
 
```
if err := s.DefaultConvert(in, out); err != nil {
  return err
}

// DeprecatedServiceAccount is an alias for ServiceAccountName.
out.DeprecatedServiceAccount = in.ServiceAccountName
```

This makes it super obvious why this is manual and not generated, and it's easy to find the custom part, and future changes will automatically just work.